cpp/include/cudf/detail/tdigest/tdigest.hpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/types.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+
+namespace cudf {
+namespace detail {
+
+// mean and weight column indices within tdigest inner struct columns
+constexpr size_type tdigest_mean_column_index   = 0;
+constexpr size_type tdigest_weight_column_index = 1;
+
+// min and max column indices within tdigest outer struct columns
+constexpr size_type tdigest_centroid_column_index = 0;
+constexpr size_type tdigest_min_column_index      = 1;
+constexpr size_type tdigest_max_column_index      = 2;
```

Comment:
Consider adding a `cudf::detail::tdigest` namespace to avoid needing to type the `tdigest_` suffix everywhere?